evm/logpoller/latency_monitor.go

```diff
@@ -0,0 +1,84 @@
+package logpoller
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-integrations/evm/types"
+)
+
+const (
+	latencyWarningThreshold = 0.7 // Warn if latency exceeds 70% of block production rate
+	latencyWarning          = "RPC latency warning: consider using faster endpoints or reviewing network conditions"
+)
+
+type LatencyMonitorClient interface {
+	HeadByNumber(ctx context.Context, n *big.Int) (*types.Head, error)
+	HeadByHash(ctx context.Context, n common.Hash) (*types.Head, error)
+}
```

Comment:
I don't think we actually care about latency for them much, because they're not going to be the botteneck. We may as well keep them there I guess, but the one that matters is `FilterLogs`.

There are two versions of it, one where the logs are requested by passing a block hash and the other where a block range is passed. It's the call to `FilterLogs` where a block hash is passed that causes problems if it's too slow.  We need to monitor this one, but shouldn't monitor the one which passes a block range since that could be much slower and that's okay.